### ISSUE TITLE
dsconf display an incomplete help with changelog setting

### ISSUE THREAD

**Issue by tbordaz** [389ds/389-ds-base#4835]
**Issue Description**
set-changelog allows duration letters like 'w', 'd'... but the online help does not mention them.
<pre>
dsconf -D "cn=directory manager" -w "password" IPA-TEST replication set-changelog --max-age 4l
Error: 103 - 3 - 53 - Server is unwilling to perform - [] - nsslapd-changelogmaxage: invalid value "4l", nsslapd-changelogmaxage must range from 0 to 9223372036854775807 or digit[sSmMhHdD]

dsconf -D "cn=directory manager" -w "password" IPA-TEST replication set-changelog --max-age 4w
Successfully updated replication changelog
</pre>

**Package Version and Platform:**
sinc 1.4.3

**Steps to Reproduce**
see desc

**Expected results**
online line should display all allowed keywords

**Comment by abeisemb** [comment]
It doesn't look like I have permissions to assign myself, but I spoke with Thierry and we confirmed that I'll be working on this.

**Comment by abeisemb** [comment]
Successfully reproduced on 389-ds-base-1.4.3, but unable to reproduce on main using build `389-ds-base-2.6.1-6.el9_6`. I created a `localhost` instance with a suffix "dc=example,dc=com" because dsconf set-changelog seems to require a valid suffix now. The given max-age parameter is accepted with no error thrown. This occurs irrespective of what I pass as a max-age: 4l, 4et, or a 20-digit long number. Is this bug expected to occur on main as well? 

```
[root@vm-10-0-186-82 ds]# dsconf -D "cn=directory manager" -w "password" localhost replication set-changelog --max-age 4l --suffix "dc=example,dc=com"
Successfully updated replication changelog
[root@vm-10-0-186-82 ds]# rpm -q 389-ds-base
389-ds-base-2.6.1-6.el9_6.x86_64
```

**Comment by tbordaz** [comment]
In 1.4.3, the tuning of the changelog was global to the server (cn=changelog5,cn=config). In 2.* it became specific to the replica. The tuned attributes are the same but not localized in the same entry.
If we provided an invalid value (e.g. '4l'), it does not fail but I have a doubt it is applied by the server. You may read back the value.
The purpose of the ticket was that if the CLI returns an error message, the message should displayed tips about the valid units ('d', 'm', 'w'..)

**Comment by abeisemb** [comment]
Right, I understand the issue with the error message--it's just curious that the error isn't printing on main. I'll submit the PR into main with the changes to the output line once it's done.

**Comment by abeisemb** [comment]
On main and RHEL 9, any invalid value is actually accepted and saved into the changelog configuration. This might have consequences down the line e.g. not pruning the changelog because nothing can ever reach that max age, but I'm not 100% sure. Investigating further. 
```
[root@vm-10-0-187-171 ~]# dsconf -D "cn=directory manager" -w "password" localhost replication set-changelog --max-age unga-bunga --suffix="dc=example,dc=com"
Successfully updated replication changelog
[root@vm-10-0-187-171 ~]# dsconf -D "cn=Directory Manager" localhost replication get-changelog --suffix "dc=example,dc=com"
dn: cn=changelog,cn=userroot,cn=ldbm database,cn=plugins,cn=config
cn: changelog
nsslapd-changelogmaxage: unga-bunga
objectClass: top
objectClass: extensibleObject



[root@vm-10-0-187-171 ~]# rpm -q 389-ds-base
389-ds-base-2.6.1-7.el9_6.x86_64
```

**Comment by abeisemb** [comment]
The input for max age should be validated in ldap/servers/plugins/replication/cl5_config.c:cldb_config_modify(), but I have been unable to find any evidence that we enter this function over the course of execution. I set up some traces in the functions dap/servers/plugins/replication/cl5_config.c:cldb_config_modify() and ldap/servers/plugins/replication/cl5_config.c:changelog5_register_config_callbacks(), but I noticed that the traces I placed in the functions weren't writing to the log files. 
At first I thought it was my mistake, but I've got traces in other functions like replica_enumerate_replicas writing successfully, which suggests that cldb_config_modify and other related functions may not be called at all--regardless of whether the new max age is valid or not.
I see cldb_config_modify called in changelog5_register_config_callbacks() and changelog5_remove_config_callbacks(), but my traces in those functions don't appear after attempting a max-age update. I think I need to trace this further.

**Comment by abeisemb** [comment]
@tbordaz I see in the [docs](https://docs.redhat.com/en/documentation/red_hat_directory_server/12/html/configuring_and_managing_replication/assembly_trimming-the-replication-changelog_configuring-and-managing-replication#proc_configuring-replication-changelog-trimming-using-the-command-line_assembly_trimming-the-replication-changelog) that the supported letters are s, m, h, d, and w. Is just adding wW to the error message considered enough, or are there more chars I should consider here?

**Comment by tbordaz** [comment]
@abeisemb the [doc](https://docs.redhat.com/en/documentation/red_hat_directory_server/12/html/configuring_and_managing_replication/assembly_trimming-the-replication-changelog_configuring-and-managing-replication#proc_configuring-replication-changelog-trimming-using-the-command-line_assembly_trimming-the-replication-changelog) is correct and covers all units. Code checking it is [here](https://github.com/389ds/389-ds-base/blob/main/ldap/servers/slapd/time.c#L716)
